db/db_impl.h

```diff
@@ -342,7 +342,8 @@ class DBImpl : public DB {
 
   using DB::EndTrace;
   virtual Status EndTrace() override;
-
+  Status TraceIteratorSeek(const uint32_t& cf_id, const Slice& key);
+  Status TraceIteratorSeekForPrev(const uint32_t& cf_id, const Slice& key);
```

Comment:
I think we should make these two functions inline.

Reply:
@siying I remember that you recently did some analysis on the benefit/non-benefit of adding explicit inline keyword. Do you think it will help here? Or would the compiler be smart enough to optimize by itself.

Reply:
We should try avoid `inline` and allow compilers to decide as much as possible. This case is a little bit tricky. If we want to do it well, I would inline the `  if (tracer_) {` part, but leave what inside the if to a separate non-inlined function. It could be:
```
Status TraceIteratorSeek(const uint32_t& cf_id, const Slice& key) {
  if (trace_) {
    TraceIteratorSeekImpl(cf_id, key);
  }
}
```